dpnp/backend/extensions/lapack/types_matrix.hpp

```diff
@@ -373,12 +373,12 @@ struct PotrfBatchTypePairSupportFactory
  * @tparam T Type of array containing input matrix A and an output arrays with
  * eigenvectors and eigenvectors.
  */
-template <typename T>
+template <typename T, typename RealT>
 struct SyevdTypePairSupportFactory
```

Comment:
We can not have a common `EvdTypePairSupportFactory` struct for `heevd` and `syevd` because they support different types

Reply:
Just an idea, not sure if it will make the overall implementation more simplified.
What if we only define type matrix with one `typename`
```
template <typename T>
struct EvdTypePairSupportFactory
{
    static constexpr bool is_defined = std::disjunction<
        dpctl_td_ns::TypePairDefinedEntry<T, double, T, double>,
        dpctl_td_ns::TypePairDefinedEntry<T, float, T, float>,
        // fall-through
        dpctl_td_ns::NotDefinedEntry>::is_defined;
};
```

And use `dispatch_vector` instead of `dispatch_table` and then update `heevd` to something similar to
```
    std::complex<T> *a = reinterpret_cast<std::complex<T> *>(in_a);
    T *w = reinterpret_cast<T *>(out_w);
```